### PR TITLE
[ARM] [AArch32] Add support for Arm China STAR-MC1 CPU

### PR DESCRIPTION
**Pull request by AlbertHuang-CPU**
STAR-MC1 is an Armv8m CPU.

Technical specifications available at:
https://www.armchina.com/download/Documents/Application-Notes/Technical-Reference-Manual?infoId=160

### CODE REVIEW

**Review by davemgreen — APPROVED** [review]
As far as I can see this LGTM. Good to see the addition.

Are you happy for the patch to be committed?

**Review by jthackray — COMMENTED** [review]
Thanks Albert. I think you might need to also update target-invalid-cpu-note/arm.c as well?

**Review by jthackray — COMMENTED** [review]
Could you add `star-mc1` to clang/test/Driver/arm-cortex-cpus-2.c?

**Review by jthackray — COMMENTED** [review]
.

File: llvm/lib/Target/ARM/ARMProcessors.td [hunk 1]
```diff
@@ -362,6 +362,16 @@ def : ProcessorModel<"cortex-m33", CortexM4Model,       [ARMv8mMainline,
...
 
+def : ProcessorModel<"star-mc1", CortexM4Model,         [ARMv8mMainline,
+                                                         FeatureDSP,
+                                                         FeatureFPARMv8_D16_SP,
+                                                         FeaturePrefLoopAlign32,
+                                                         FeatureHasSlowFPVMLx,
+                                                         FeatureHasSlowFPVFMx,
+                                                         FeatureUseMISched,
+                                                         FeatureHasNoBranchPredictor,
+                                                         FeatureFixCMSE_CVE_2021_35465]>;
```
[the remark this answers is not in the document]
Comment (AlbertHuang-CPU) [reply llvm/lib/Target/ARM/ARMProcessors.td:373]: Thanks Jonathan. The two files updated and the changes passed the Unit tests of clang and llvm.

**Review by jthackray — APPROVED** [review]
Thanks, LGTM.

**Comment by davemgreen** [comment]
Can you rebase? FeaturePrefLoopAlign32 has been renamed. Then if you are happy for this to be committed then let us know and we can hit the squash and merge button (it is not always obvious who does and does not have commit access).

**Comment by jthackray** [comment]
> Hi David, rebase done. Could you help to merge?

It doesn't look like it builds at the moment. Happy to help merge when it's ready :)

**Pull request merged**